### ISSUE TITLE
[New Metric] Hinge losses

### ISSUE THREAD

**Issue by SkafteNicki** [PyTorchLightning/metrics#89]
## 🚀 Feature

Add Hinge losses to classification package:
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.hinge_loss.html#sklearn.metrics.hinge_loss
possibly with an added parameter `squared` to also calculated squared hinge loss

**Comment by ethanwharris** [comment]
Hi @SkafteNicki, I'd like to take a look at adding `hinge_loss` to `torchmetrics` :) Could you assign me to the issue?

**Comment by SkafteNicki** [comment]
@ethanwharris Nice, thanks for wanting to contribute :]
